quick-start.sh

```diff
@@ -7,169 +7,347 @@
+  BILLING_ENABLED=$(gcloud beta billing projects describe "${PROJECT_ID}" --format="value(billingEnabled)")
+  if [[ "${BILLING_ENABLED}" == "False" ]]; then
+    enable_billing
+    link_billing_account
+  fi
+else
+  NEW_PROJECT=true
+  echo "Thank you! We will now attempt to create a new Google Cloud $(pink 'project') for you."
+  echo "A window will open asking you to authorize the gcloud CLI. Please click '$(pink 'Authorize')'."
+  echo
+
+  # Check if billing is enabled, enable billing if needed
+  enable_billing
+
+  PROJECT_NAME=$(gum input --prompt="Please enter a name for a new $(pink 'Project'). " --placeholder="Project name")
+  PROJECT_ID=$(echo $PROJECT_NAME | awk '{print tolower($0)}')-$(date +"%s")
+  spin "Creating $(pink 'project')..." gcloud projects create $PROJECT_ID --name="${PROJECT_NAME}"
+
+  # Link billing account to project
+  link_billing_account
 fi
 
 # Set the current project to the PROJECT_ID specified above
-gcloud config set project "${PROJECT_ID}"
+spin "Setting gcloud default $(pink 'project')..." gcloud config set project "${PROJECT_ID}"
 
 # Enable necessary APIs
-gcloud services enable \
-    iam.googleapis.com \
-    cloudresourcemanager.googleapis.com \
-    iamcredentials.googleapis.com \
-    sts.googleapis.com \
-    serviceusage.googleapis.com
+spin "Enabling $(pink 'gcloud APIs') (this may take a few minutes)..." gcloud services enable \
+      compute.googleapis.com \
+      iam.googleapis.com \
+      cloudresourcemanager.googleapis.com \
+      iamcredentials.googleapis.com \
+      sts.googleapis.com \
+      serviceusage.googleapis.com
+
+# Prompt for region, zone, and Smarty creds
+echo "Please select the $(pink 'region') you would like to deploy to."
+echo "More info: https://cloud.google.com/compute/docs/regions-zones/regions-zones"
+echo
+REGION=$(gcloud compute regions list --filter="name~'us-'" --format="value(name)" | gum choose)
+
+echo "Please select the $(pink 'zone') you would like to deploy to."
+echo
+ZONE=$(gcloud compute zones list --filter="name~'${REGION}'" --format="value(name)" | gum choose)
+
+echo "Please enter the $(pink 'Authorization ID') of your Smarty Street Account."
+echo "More info: https://www.smarty.com/docs/cloud/authentication"
+echo
+SMARTY_AUTH_ID=$(gum input --placeholder="Authorization ID")
+
+echo "Please enter the $(pink 'Authorization Token') of your Smarty Street Account."
+echo "More info: https://www.smarty.com/docs/cloud/authentication"
+echo
+SMARTY_AUTH_TOKEN=$(gum input --placeholder="Authorization Token")
+
+# Login to gh CLI
+echo "Project ID $(pink 'set')!"
+echo "We will now login to the $(pink 'GitHub CLI')."
+echo "Copy the provided code, press $(pink 'Enter') and then click the link that will be printed."
+echo "Paste the code into the browser tab that opens and follow the prompts to authorize GitHub."
+echo "Then return to this terminal!"
+echo
+
+gh auth login --hostname github.com -p https -w
+GITHUB_USER=$(gh api user -q '.login')
```

Comment:
I asked this question before, but think we may have had a misunderstanding. 

I'm curious about what happens if the user isn't already signed into their GH account via their browser. @nickclyde, I believe you previously mentioned that they will either be prompted to auth or re-auth, but did you mean this in regard to signing in via the command line? I'm wondering if we need to ask the user to make sure they are signed into GH via the browser as part of this step. Thoughts?

Reply:
If this is what happens, could we also add some logging for the user explaining that they may first need to log into GH before pasting the access code if they are not already signed in?

Reply:
Just confirmed that that is the actual flow! Added some additional logging as well. :+1: